packages/widget/src/components/GroupedAssetImage.tsx

```diff
@@ -0,0 +1,52 @@
+import { GroupedAsset } from "@/modals/AssetAndChainSelectorModal/AssetAndChainSelectorModal";
+import { useState } from "react";
+import styled from "styled-components";
+
+const NUMBER_OF_IMAGES_TO_CHECK = 6;
+
+export type GroupedAssetImageType = {
+  groupedAsset?: GroupedAsset;
+  height?: number;
+  width?: number;
+};
+
+export const GroupedAssetImage = ({ groupedAsset, height, width }: GroupedAssetImageType) => {
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
+
+  if (!groupedAsset?.assets) return <img height={height} width={width} />;
+  const allLogoURIs = [
+    groupedAsset.assets.find((asset) => asset.logoURI?.includes("raw.githubusercontent"))?.logoURI,
+    ...groupedAsset.assets.map((asset) => asset.logoURI),
+  ].filter((uri): uri is string => !!uri);
+
+  const dedupedLogoURIs = Array.from(new Set(allLogoURIs));
```

Comment:
in practice after deduping the logos there's usually not that many, typically 1-2 so we aren't attempting to load a whole bunch of images, but to be safe I added a hard limit of 6